src/memory-storage/index.ts

```diff
@@ -0,0 +1,61 @@
+import {
+    AccessToken as CoreAccessToken,
+    User as CoreUser,
+} from '../models';
+import Stored from './storage';
+
+export class AccessToken extends
```

Comment:
I think we need to add a kind of jsdoc or comments to our classes too, this way when someone extend the class, they IDE will tell them what each parameter e.g. `consumed` are.

Reply:
@hesamd I added lots of more detailed documentations.
Unfortunately I wasn't able to document destructured parameters:

https://github.com/TypeStrong/typedoc/issues/653
https://github.com/TypeStrong/typedoc/issues/522
https://github.com/TypeStrong/typedoc/issues/127
https://github.com/TypeStrong/typedoc/issues/957

So we have two options:
1. Write it as raw text (which is really annoying and needs to update after changes and may make maintenance hard.)
2. Help typedoc project to have this feature by providing a PR there.

Reply:
In the interim, can we go with the suggestion of [this guy](https://github.com/TypeStrong/typedoc/issues/653#issuecomment-351611439)